language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -242,32 +298,57 @@ substitution, anything).
 -- 2. calls 'returnCek' on values ('TyAbs', 'LamAbs', 'Constant')
 -- 3. returns 'EvaluationFailure' ('Error')
 -- 4. looks up a variable in the environment and calls 'returnCek' ('Var')
+
+
+getArgsCount :: Builtin ann -> CekM uni Int
+getArgsCount (BuiltinName _ name) =
+    pure $ builtinNameArities ! name
+getArgsCount (DynBuiltinName _ name) = do
+    DynamicBuiltinNameMeaning sch _ _ <- lookupDynamicBuiltinName name
+    pure $ countArgs sch
```

Comment:
Quoting a slack message:

> If we have a map of dynamic builtins, then we can simply put the precomputed arity of each of the builtins there and we won't even need to compute them in the main evaluation pipeline

Reply:
I'm supposed to be looking at getting rid of static builtins altogether soon, so maybe I'll defer that until then.  At the moment we're not using many dynamic builtins, so it's not critical.  I'll add a comment though.

Maybe it's not so hard though: I'll take a look on Monday.